### ISSUE TITLE
Parsing bug in 02/22 nightly

### ISSUE THREAD

**Issue by nipunn1313** [rust-lang/rust#58962]
Haven't confirmed if it has been fixed since 02/22

```
pub fn f<F: FnOnce(u64) -> u64>(f1: F) -> u64 {
        f1(1)
}

#[test]
fn test_weird_rustc_thing() {
    let y = f({
        |x| x+1
    ); // shouldn't this need a closing curly brace?
    assert_eq!(y, 2);
    println!("WEIRD RUSTC BUG");
}
```

Seems to only repro inside an inner mod.

Here's a link to a repro
https://github.com/nipunn1313/rustc_parser_bug

**Comment by estebank** [comment]
Introduced in #57944. Will be fixed in #58903. Duplicate of #58886.

**Comment by nipunn1313** [comment]
Thanks! #57944 looks pretty great. Especially given the test case on the fix https://github.com/rust-lang/rust/pull/58903/files#diff-44324609d226c76463a9b78f62aba74d !!
